tower/src/retry/budget.rs

```diff
@@ -89,21 +89,51 @@ use tokio::time::Instant;
 /// For more info about [`Budget`], please see the [module-level documentation].
 ///
 /// [module-level documentation]: self
-pub struct Budget {
-    bucket: Bucket,
+pub struct Budget<B = TpsBucket> {
```

Comment:
So I would not add a generic here and instead move this whole file to a new submodule and then have the main trait in the mod file.